### PR TITLE
Use pgx

### PR DESCRIPTION
**Pull request by dehume**
Use pgx instead of lib/pq for the connection.

### CODE REVIEW

File: materialize/provider.go [hunk 1]
```diff
@@ -10,8 +10,8 @@ import (
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	_ "github.com/jackc/pgx/stdlib"
 	"github.com/jmoiron/sqlx"
-	_ "github.com/lib/pq" //PostgreSQL db
```
Comment (benesch) [inline materialize/provider.go:14]: I think this is getting pulled in somewhere else too, because it's not showing up as removed in go.mod.

Comment (dehume) [reply materialize/provider.go:14]: Looks like it is being used for `pq.QuoteLiteral` in the secrets. Looks like this was the suggested solution in https://github.com/MaterializeInc/terraform-provider-materialize/pull/18. Is there a better alternative that doesn't use pg?

Comment (benesch) [reply materialize/provider.go:14]: Yes! I linked the equivalent pgx functions here: https://github.com/MaterializeInc/terraform-provider-materialize/issues/27#issue-1590190068

**Review by bobbyiliev — APPROVED** [review]
This looks good to me! Leaving it up to @benesch to confirm if he has any other concerns.

**Review by benesch — COMMENTED** [review]
This is great, thanks! (Sorry, I've not been fastidious with keeping track of PRs in this repo—feel free to ping me on Slack if you need my eyes on something!)

**Pull request merged**